docs/how-to-guides/add-a-custom-ros-message.md

```diff
@@ -0,0 +1,127 @@
+
+The following is a simple tutorial of adding a `autoware_msgs` to Autoware, for the general ROS2 tutorial see [Create custom msg and srv files](http://docs.ros.org/en/galactic/Tutorials/Beginner-Client-Libraries/Custom-ROS2-Interfaces.html).
+
+## How to create custom message
+
+Make sure you are in the Autoware workspace, and then run the following command to create a new package.
+For example we create a package to define sensor messages.
+
+1. Create a package
+
+   ```console
+   cd ./src/core/autoware_msgs
+   ros2 pkg create --build-type ament_cmake autoware_sensing_msgs
+   ```
+
+2. Create custom messages
+
+   You should create `.msg` files and place them in the `msg` directory.
+
+   **NOTE**: The initial letters of the `.msg` and `.srv` files must be capitalized.
+
+   For example we make msg files `GnssInsOrientation.msg` and `GnssInsOrientationStamped.msg` to define GNSS INS orientation messages:
+
+   ```console
+   mkdir msg
+   cd msg
+   touch GnssInsOrientation.msg
+   touch GnssInsOrientationStamped.msg
+   ```
+
+   The `GnssInsOrientation.msg` with the following content:
+
+   ```c++
+   geometry_msgs/Quaternion orientation
+   float32 rmse_rotation_x
+   float32 rmse_rotation_y
+   float32 rmse_rotation_z
+   ```
+
+   In this case, the custom message uses a message from another message package `geometry_msgs/Quaternion`.
+
+   The `GnssInsOrientationStamped.msg` with the following content:
+
+   ```c++
+   std_msgs/Header header
+   GnssInsOrientation orientation
+   ```
+
+   In this case, the custom message uses a message from another message package `std_msgs/Header`.
+
+3. CmakeLists.txt
+
+   In order to use this custom message in `C++` or `Python` languages, we need add the following lines to `CmakeList.txt`:
+
+   ```cmake
+   rosidl_generate_interfaces(${PROJECT_NAME}
+     "msg/GnssInsOrientation.msg"
+     "msg/GnssInsOrientationStamped.msg"
+     DEPENDENCIES
+       geometry_msgs
+       std_msgs
+     ADD_LINTER_TESTS
+   )
+   ```
+
+   :speech_balloon: The `ament_cmake_auto` tool is very useful and is more widely used in Autoware, so we recommend using `ament_cmake_auto` instead of `ament_cmake`.
+
+   We need to replace
+
+   ```cmake
+   find_package(ament_cmake REQUIRED)
+
+   ament_package()
+   ```
+
+   with
+
+   ```cmake
+   find_package(ament_cmake_auto REQUIRED)
+
+   ament_auto_package()
+   ```
+
+4. package.xml
+
+   We need to declare relevant dependencies in `package.xml`. For the above example we need to add the following content:
+
+   ```xml
+   <depend>geometry_msgs</depend>
+   <depend>std_msgs</depend>
+   <buildtool_depend>rosidl_default_generators</buildtool_depend>
+   <exec_depend>rosidl_default_runtime</exec_depend>
+   <member_of_group>rosidl_interface_packages</member_of_group>
```

Comment:
To keep consistent style, it might be better to sort lines by the order written in http://download.ros.org/schema/package_format3.xsd.
![image](https://user-images.githubusercontent.com/31987104/208952065-3e600e6a-e97c-475c-b675-8f26231d8422.png)

```suggestion
   <buildtool_depend>rosidl_default_generators</buildtool_depend>
   
   <exec_depend>rosidl_default_runtime</exec_depend>

   <depend>geometry_msgs</depend>
   <depend>std_msgs</depend>

   <member_of_group>rosidl_interface_packages</member_of_group>
```